### DISCUSSION TITLE
Test of subroutines and modal state save/restore (M70-M73)

### DISCUSSION THREAD

**Discussion by terjeio** [grblHAL/core#516] [Ideas]
Code for test of patch for nested/recursive calls (G65 and o<n> CALL), [LinuxCNC subroutines](https://linuxcnc.org/docs/html/gcode/o-code.html) and [saving/restoring of modal state](http://linuxcnc.org/docs/stable/html/gcode/m-code.html#mcode:m70). Related information can be found in [this discussion](https://github.com/grblHAL/core/discussions/309).

Copy over latest build (20240513) and recompile.
I have only done some light testing so be careful when trying out a patched controller.

[expr_patch.zip](https://github.com/grblHAL/core/files/15351372/expr_patch.zip)

**Comment by andrewmarles** [comment]
Here's a bunch of FlexiHAL builds with the patch applied.
[FlexiHAL_FW_expr_patch.zip](https://github.com/grblHAL/core/files/15353208/FlexiHAL_FW_expr_patch.zip)

**Comment by mintracer** [comment]
i´m getting compiling errors when compiling for the grblhal2k-EST - DRIVER:iMXRT1062 - patch is applied to the core with the date: [VER:1.1f.20240513:]

![Screenshot 2024-05-18 001740](https://github.com/grblHAL/core/assets/97637453/9bcd8112-12e2-4e4f-9345-be9cac4d3333)

**Reply by mintracer** [reply]
solved... --> line 521 in /grbl/config.h -> #define NGC_EXPRESSIONS_ENABLE On was Off
